[RFC] vim-patch:7.4.{984,1093}

watiko opened the pull request:
#### vim-patch:7.4.984

Problem:    searchpos() always starts searching in the first column, which is
            not what some people expect. (Brett Stahlman)
Solution:   Add the 'z' flag: start at the specified column.

https://github.com/vim/vim/commit/ad4d8a192abf44b89371af87d70b971cd654b799
#### tests: Migrate legacy test searchpos
#### vim-patch:7.4.1093

Problem:    Typo in test goes unnoticed.
Solution:   Fix the typo. Give error for wrong arguments to cursor().
            (partly by Hirohito Higashi) Add a test for cursor().

https://github.com/vim/vim/commit/24c4d539eed33e8073f8f9fe2bee497bbba935a4

watiko committed:
vim-patch:7.4.984

Problem:    searchpos() always starts searching in the first column, which is
            not what some people expect. (Brett Stahlman)
Solution:   Add the 'z' flag: start at the specified column.

https://github.com/vim/vim/commit/ad4d8a192abf44b89371af87d70b971cd654b799

watiko committed:
tests: Migrate legacy test searchpos

cacplate commented on src/nvim/search.c:
```diff
@@ -453,25 +453,24 @@ void last_pat_prog(regmmatch_T *regmatch)
+//
+// Return FAIL (zero) for failure, non-zero for success.
+// Returns the index of the first matching
+// subpattern plus one; one if there was none.
```
This could be doxygen style since you already change it.

oni-link commented on runtime/doc/eval.txt:
```diff
@@ -5427,14 +5427,15 @@ search({pattern} [, {flags} [, {stopline} [, {timeout}]]])	*search()*
+		's'	Set the ' mark at the previous location of the cursor
+		'w'	Wrap around the end of the file
+		'W'	don't Wrap around the end of the file
+		'z'	start searching at the cursor column instead of zero
```
Zero instead of zero?

watiko replied:
I don't think so. `Zero` should be used the first word of sentence.

oni-link replied:
The other changes here do the same to help with the meaning of the flag character.
`'b'` -> Backward
`'c'` -> Cursor
...
`'z'` -> Zero

watiko replied:
Ah, I see. I will do it.

BTW, will you send it to vim_dev?

oni-link commented on src/nvim/search.c:
```diff
@@ -571,16 +570,15 @@ int searchit(
-         */
+        // Look for a match somewhere in line "lnum".
+        colnr_T col = at_first_line && (options & SEARCH_COL)
+                      ? pos->col : (colnr_T)0;
```
`(colnr_T) 0` -> `0`, also fits in one line.

oni-link commented on src/nvim/version.c:
```diff
@@ -197,7 +197,7 @@ static int included_patches[] = {
   // 1095 NA
   // 1094,
-  // 1093,
+  // 1093 NA
```
Why is this NA? `f_cursor()`in eval.c is also modified (error message, function comment).

watiko replied:
Oh, I missed it...

watiko committed:
vim-patch:7.4.1093

Problem:    Typo in test goes unnoticed.
Solution:   Fix the typo. Give error for wrong arguments to cursor().
            (partly by Hirohito Higashi) Add a test for cursor().

https://github.com/vim/vim/commit/24c4d539eed33e8073f8f9fe2bee497bbba935a4

watiko commented:
Updated. Thanks: @cacplate @oni-link
